Initial implementation of authentication flow

MudassirQuraishi opened the pull request:
This commit introduces the initial setup and basic functionality for the authentication flow. Included are login, and signup, along with necessary backend configurations.

MudassirQuraishi committed:
Initial implementation of authentication flow

This commit introduces the initial setup and basic functionality for the authentication flow. Included are login, and signup, along with necessary backend configurations.

MudassirQuraishi committed:
Changes made in Auth flow

Updated the Labels for signup , login forms, included icon for logo

sourav0809 commented on src/api/api.js:
add some comments for apis

MudassirQuraishi replied:
Lets do documentation at the end.

MudassirQuraishi replied:
Added documentation for API

sourav0809 commented on src/app/App.jsx:
```diff
@@ -0,0 +1,27 @@
+    return (
+        <>
+            {loading && <PageLoader />}
+            {!isLoggedIn && <AuthRoutes />}
```
check is logged in in single line use the ternary operator

MudassirQuraishi replied:
Updated the logic here

sourav0809 commented on src/components/auth/AuthHeader.jsx:
```diff
@@ -0,0 +1,14 @@
+        </div>
+    </>)
+}
+AuthHeader.propTypes = {
```
don't need this

MudassirQuraishi replied:
Removed Proptypes

sourav0809 commented on src/components/auth/AuthLayout.jsx:
```diff
@@ -0,0 +1,57 @@
+import { CalendarCheck } from 'lucide-react'
+const AuthLayout = ({ children }) => {
+    const [state, setState] = useState(false)
+    const menus = [
```
make a separate file called header constant

MudassirQuraishi replied:
Moved the data to another file

sourav0809 commented on src/components/auth/BackButton.jsx:
```diff
@@ -0,0 +1,17 @@
+
+    </>)
+}
+BackButton.propTypes = {
```
remove this

MudassirQuraishi replied:
Removed proptypes

sourav0809 commented on src/components/ui/button.jsx:
```diff
@@ -0,0 +1,47 @@
+
+import { cn } from "@/lib/utils"
+
+const buttonVariants = cva(
```
should start with capital letter

MudassirQuraishi replied:
These are shadcn's components,

MudassirQuraishi replied:
Updated file names

sourav0809 commented on src/components/auth/Login.jsx:
```diff
@@ -0,0 +1,80 @@
+    }
+    return (<>
+        <CardWrapper label="Welcome back, we missed you." title="Login" backButtonPath="/auth/signup" backButtonLabel="Don't have an account? Signup here">
+            <Form {...form}>
```
if possible try to make your own form

MudassirQuraishi replied:
We have the option to use a UI Library

sourav0809 commented on src/components/common/CardWrapper.jsx:
```diff
@@ -0,0 +1,40 @@
+};
+
+
+CardWrapper.propTypes = {
```
remove this

MudassirQuraishi replied:
Removed Proptypes

sourav0809 commented on src/routes/AuthRoutes.jsx:
```diff
@@ -0,0 +1,17 @@
+            <Routes>
+                <Route path="/auth/signup" element={<Signup />} />
+                <Route path="/auth/login" element={<Login />} />
+            </Routes>
```
no error routes present add "*" in the last

sourav0809 commented on src/routes/UserRoutes.jsx:
```diff
@@ -0,0 +1,10 @@
+    return (<>
+        <Routes>
+            <Route path="/home" />
+        </Routes>
```
no error route present

sourav0809 commented on src/schema/index.js:
```diff
@@ -0,0 +1,28 @@
+import { z } from 'zod';
+export const signupSchema = z.object({
```
change the name

sourav0809 commented on src/store/actions/authActions.js:
```diff
@@ -0,0 +1,56 @@
+  setUserLoggedIn,
+  logOutUser,
+} from '../reducers/authSlice';
+
```
add some comments

MudassirQuraishi replied:
Better comments added

sourav0809 commented on src/store/actions/authActions.js:
```diff
@@ -0,0 +1,56 @@
+    } catch (err) {
+      console.log(err);
+    }
+    setBtnLoader(false);
```
make the button loader false in finally block

MudassirQuraishi replied:
update the try catch blocks

MudassirQuraishi committed:
Documentation added

Documentation added for API and authActions

MudassirQuraishi committed:
Resolved Issues for initial auth flow

Removed PropTypes for components, updated component names and imports, created a new file for menu links

MudassirQuraishi committed:
Routes Updated

Handled unexpected routes, and redirected them to proper routes.
